source/actors/Jump.gd

```diff
@@ -8,18 +10,23 @@ onready var _available_jumps = max_jumps
 
 
 func _physics_process(delta):
-	if plaform_actor.is_on_floor():
+	if platform_actor.is_on_floor():
 		_available_jumps = max_jumps
-		plaform_actor.snap_normal = Vector2.DOWN
+		platform_actor.snap_normal = Vector2.DOWN
+		emit_signal("available_jumps_replenished")
+		set_physics_process(false)
 
 
 func apply():
-	if _available_jumps > 0:
-		plaform_actor.velocity.y = -strength
+	if _available_jumps > 0 or max_jumps == -1:
```

Comment:
here it is safer adding a `<=-1` instead of `==` even if this condition should never happened. 

But I think all these hardcoded numbers make the code hardly understandable. I would go with maybe nested condition and enum that describes behaviours

Reply:
scratch the first part, this is definitely a case of enum